libs/checkpoint/src/types.ts

```diff
@@ -36,3 +36,26 @@ export interface CheckpointMetadata {
    */
   parents: Record<string, string>;
 }
+
+const checkpointMetadataKeys = ["source", "step", "writes", "parents"] as const;
```

Comment:
I think we might accept arbitrary metadata keys?

Reply:
I was just taking cues from `PostgresSaver` on this. Is there some mechanism for user code to modify the checkpoint metadata? It doesn't have an index signature, so I (perhaps incorrectly) assumed it was a complete type that originated internally. The typing cues are contradictory though, as `options.filter` is `Record<string, unknown>`. A type of `Partial<CheckpointMetadata>` would've been a clearer expression of intent there if it was indeed meant to be restricted.

Regardless, I added it in the RDBMS implementations because the list of filter keys is being directly concatenated into the query text (SQL injection potential), and as a result I also added it here for consistency.

Assuming arbitrary filter keys _aren't_ desired, perhaps it would be better to throw on unknown keys. Silently dropping them could be confusing behavior, otherwise.

If they _are_ desired, we should think carefully about how to sanitize the inputs for the RDBMS implementations.